server/src/handler_monitor.c

```diff
@@ -53,84 +53,79 @@ onion_connection_status handler_get_trains(void *_, onion_request *req,
 			                       train_grabbed(query.ids[i]) ? "yes" : "no");
 		}
 		bidib_free_id_list_query(query);
-		char response[trains->len + 1];
-		strcpy(response, trains->str);
+		onion_response_printf(res, "%s", trains->str);
+		syslog_server(LOG_INFO, "Request: Get available trains - finished");
```

Comment:
I thought "- finished" is only for handlers that do actual processing?

Reply:
So far, " - finished " is for handlers which have more than one or two lines roughly; i.e. for those where some error (failed allocation of the glib string etc.) could occur. In contrast, if all the handler does is set one bool, I don't think any error could possibly occur because of it.    
However, yes, we could think about doing " - done" instead for all requests that are "just" getters.